### PR TITLE
Fix many typos in docs and comments

### PR DESCRIPTION
**Pull request by mpociecha**
This PR corrects many typos found in scaladocs, comments and
documentation. It should reduce a bit number of PRs which fix one
typo.

There are no changes in the 'real' code except one corrected name of
a JUnit test method and some error messages in exceptions. In the case
of typos in other method or field names etc., I just skipped them.

Obviously this commit doesn't fix all existing typos. I just generated
in IntelliJ the list of potential typos and looked through it quickly.

### CODE REVIEW

File: src/repl/scala/tools/nsc/interpreter/JavapClass.scala [hunk 1]
```diff
@@ -743,7 +743,7 @@ object JavapClass {
             case _                  => Nil
           }
       val res = fs map (_.to[Seq]) getOrElse Seq()
-      // on second thought, we don't care about lamba method classes, just the impl methods
+      // on second thought, we don't care about lambda method classes, just the impl methods
       val rev =
       res flatMap {
         case x @ closure(_, "lambda", _, _) => labdaMethod(x, target)
```
Comment (som-snytt) [inline src/repl/scala/tools/nsc/interpreter/JavapClass.scala:749]: So glad you skipped this intentional spelling.

**Comment by som-snytt** [comment]
LGTM

Do the corrections to "overridden" include usages of "overwritten"?  Just kidding.

File: src/repl/scala/tools/nsc/interpreter/JavapClass.scala (hunk 1 above)
Comment (mpociecha) [inline src/repl/scala/tools/nsc/interpreter/JavapClass.scala:749]: It's a public method of a public class - in general I was avoiding the changes which could (let's say) break the compatibility etc. I'm sure that then I would have to explain why I changed such things and why it's safe and so on, and so on.

Edit: Or could I just create new methods / classes with corrected names and still have the old ones but as deprecated methods / type aliases redirecting to these newly created ones?

**Comment by mpociecha** [comment]
@som-snytt I changed also src\compiler\scala\tools\nsc\transform\Erasure.scala, src\partest-extras\scala\tools\partest\BytecodeTest.scala and test\benchmarks\src\scala\collection\parallel\benchmarks\Bench.scala (if you'd like to check them ;) ) And don't worry - I won't add anything more.

**Comment by som-snytt** [comment]
I'm sure they're fine. Just comments, right? I would have made it a second commit. Also, at a certain size, github no longer shows complete diffs.

**Comment by mpociecha** [comment]
Yes, only comments. And github shows up to 300 files so it's still far from that. :)

**Comment by gkossakowski** [comment]
Cool! Thanks a lot for following [The Boy Scout Rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule)!

**Pull request merged**